zio-http/src/main/scala/zio/http/URL.scala

```diff
@@ -101,7 +101,7 @@ final case class URL(
 
   def host(host: String): URL = {
     val location = kind match {
-      case URL.Location.Relative      => URL.Location.Absolute(Scheme.HTTP, host, URL.portFromScheme(Scheme.HTTP))
+      case URL.Location.Relative      => URL.Location.Absolute(Scheme.HTTP, host, Scheme.defaultPortForHTTP)
```

Comment:
The port in `Location` is publicly accessible. Therefore I think we should not have a `-1` here. I'd rather make port an `Option` or have two variants of `Absolute`, one with and one without port.

Reply:
I did consider making the port an Option.
ok I would start changing some APIs (methods signature)

Reply:
Done